File: Java/26-RemoveDuplicatesFromSortedArray.java
public class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums == null || nums.length == 0)
            return 0;
        int slow = 1;
        int fast = 1;
        while (fast < nums.length){
            if (nums[fast] == nums[fast-1]){
                fast++;
            }else{
                nums[slow] = nums[fast];
                slow++;
                fast++;
            }
        }
        return slow;
    }
    //Second solution
    public int removeDuplicates2(int[] nums) {
        if (nums == null || nums.length == 0){
            return 0;
        }
        int p = 1, len = nums.length;
        for (int i = 1; i < len; i++){
            if (nums[i] != nums[i-1]){
                nums[p] = nums[i];
                p++;
            }
        }
        return p;
    }
}

File: cpp/26-RemoveDuplicatesFromSortedArray.cpp
//[1,1,2,2,3,3,4,4,5,6,7]
//Not swap!
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() == 0) {
            return 0;
        }
        int p1 = 1, p2 = 1, len = nums.size();
        while (p1 < len && p2 < len) {
            if (nums[p2] == nums[p2-1]) {
                p2++;
            } else {
                nums[p1] = nums[p2];
                p1++;
                p2++;
            }
        }
        return p1;
    }
};
